include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -840,16 +867,32 @@ struct __get_zeroth_element
         return std::get<0>(std::forward<_Tp>(__a));
     }
 };
-
+template <typename Assign = oneapi::dpl::__internal::__pstl_assign>
 struct __write_to_idx_if
 {
     template <typename _OutRng, typename _SizeType, typename ValueType>
     void
     operator()(_OutRng&& __out, _SizeType __idx, const ValueType& __v) const
     {
         if (std::get<1>(__v))
-            __out[std::get<0>(__v) - 1] = std::get<2>(__v);
+            __assign(std::get<2>(__v), __out[std::get<0>(__v) - 1]);
+    }
+    Assign __assign;
+};
+
+template <typename Assign = oneapi::dpl::__internal::__pstl_assign>
+struct __write_to_idx_if_else
+{
+    template <typename _OutRng, typename _SizeType, typename ValueType>
+    void
+    operator()(_OutRng&& __out, _SizeType __idx, const ValueType& __v) const
+    {
+        if (std::get<1>(__v))
```

Comment:
We will likely need a fix similar to https://github.com/oneapi-src/oneDPL/pull/1714 here.

Reply:
yes, agreed.  I'll rebase and make the appropriate changes.

Reply:
should be done.